File: src/routes/analyze.js
import { Router } from 'express';
import { analyzeStudent } from '../services/analyzer.js';

const router = Router();

router.post('/:studentId', async (req, res) => {
  try {
    const result = await analyzeStudent(req.params.studentId);
    if (!result) return res.status(404).json({ error: 'Student not found' });
    res.json(result);
  } catch (err) {
    console.error(err);
    res.status(500).json({ error: err.message });
  }
});

export default router;

File: app/routers/analyze.py
from fastapi import APIRouter
from app.services.analyzer import analyze_student

router = APIRouter()

@router.post("/analyze/{student_id}")
def analyze(student_id: str):
    result = analyze_student(student_id)
    if not result:
        return {"error": f"Student {student_id} not found"}
    return result